### PR TITLE
Add `StorefrontGetCustomFieldCollection`

### PR DESCRIPTION
**Pull request by justnero**
## Summary

Only a subset of custom fields supported resource types would be available in Storefront. 

## Checklist

- [x] Writing style
- [x] API design standards

### CODE REVIEW

File: openapi/paths/storefront/custom-fields@{resource}.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+parameters:
+  - $ref: ../../components/parameters/storefrontCustomFieldResource.yaml
```
Comment (jvc9109) [inline openapi/paths/storefront/custom-fields@{resource}.yaml:2]: I wonder if it should referenced in the same place as `collectionLimit` and `collectionOffset` parameters.

Comment (justnero) [reply openapi/paths/storefront/custom-fields@{resource}.yaml:2]: No, this one is a URL parameter that applies to all methods below (even if there's only one right now).

**Review by jvc9109 — APPROVED** [review]

**Comment by ashkarpetin** [comment]
Why `Writing style` unchecked?

**Comment by justnero** [comment]
@ashkarpetin missed it, thank you for noticing.

**Review by ashkarpetin — APPROVED** [review]

**Review by nanobash — APPROVED** [review]

**Review by ninjasimon — APPROVED** [review]

**Pull request merged**